src/main/java/org/reaktivity/nukleus/http_cache/util/Slab.java

```diff
@@ -0,0 +1,162 @@
+/**
+ * Copyright 2016-2017 The Reaktivity Project
+ *
+ * The Reaktivity Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package org.reaktivity.nukleus.http_cache.util;
+
+import static org.agrona.BitUtil.isPowerOfTwo;
+
+import java.nio.ByteBuffer;
+import java.util.BitSet;
+
+import org.agrona.MutableDirectBuffer;
+import org.agrona.collections.Hashing;
+import org.agrona.concurrent.UnsafeBuffer;
+import org.reaktivity.nukleus.buffer.BufferPool;
+
+/**
+ * A chunk of shared memory for temporary storage of data. This is logically segmented into a set of
+ * slots of equal size. Methods are provided for acquiring a slot, getting a slabBuffer that can be used
+ * to store data in it, and releasing the slot once it is no longer needed.
+ * <b>Each instance of this class is assumed to be used by one and only one thread.</b>
+ */
+public class Slab implements BufferPool
```

Comment:
Please mark this as `@Deprecated` as a reminder that this will be replaced with a cache-specific approach later.